WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPLocaleTestRule.java

```diff
@@ -1,57 +1,61 @@
 package org.wordpress.android.ui.screenshots;
 
+import android.content.Context;
 
 import androidx.test.core.app.ApplicationProvider;
-import androidx.test.platform.app.InstrumentationRegistry;
 
 import org.junit.rules.TestRule;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 import org.wordpress.android.WordPress;
 import org.wordpress.android.util.LocaleManager;
 
-public class WPLocaleTestRule implements TestRule {
-    private static final String FASTLANE_TEST_LOCALE_KEY = "testLocale";
-    private static final String FASTLANE_ENDING_LOCALE_KEY = "endingLocale";
+import java.util.Locale;
+
+import tools.fastlane.screengrab.locale.LocaleUtil;
 
+public class WPLocaleTestRule implements TestRule {
```

Comment:
Question (❓): The below was copied from the commit message itself, that is, part of the description on it:

```
But I still think it's valuable to add our `WPLocaleTestRule` to the `RuleChain`
for the case when developers work on fixing the screenshot UI tests and might
try to run them from Android Studio, via an emulator they might not have
wiped out first, and on which they might already have the APK installed from
previous work, and in which they might have changed the in-app locale.
So better safe than sorry :)
```

I want to challenge this a bit in favor of removing our custom `WPLocaleTestRule` no matter. However, FYI, I have no strong opinions anyway, I just want to challenge this in favor of ending-up with less code/configuration to maintain for later on.

As such, IMHO, for the case when engineers work on fixing the screenshots UI tests and might run them from AS, I think it is better to document that case and make it explicit, so that engineers would always wipe out the emulator first or uninstall the APK first. I think it is better to have a strict and documented process on something that would rarely be the case, than having a custom rule helping us overcome it, in cases where engineers will need it, just because they missed to read the clear instructions on how to use the screenshot solution. Wdyt?

PS: By removing the `WPLocaleTestRule`, apart from the obvious code/configuration wins, we can get rid of the `WordPress.updateContextLocale(...)` in favor of using the `AppInitializer.updateContextLocale()` direct within the `AppSettingsFragment.changeLanguage(...)` method, it being the only usage afterwards, as well as reverting it to become as it was, that is, without it accepting the extra `appContext` parameters.

Reply:
That's an interesting take.

So basically, remove the `WPLocaleTestRule`, and add a comment in the code of the test classes to tell developers to ensure they uninstall any previous version of the app before testing to ensure that the in-app locale switched and user preference for overridden locale does not accidentally take over the device locale set by the `LocaleTestRule` from fastlane?

I'm ok with that tbh. Basically that fact of keeping `WPLocaleTestRule` came to be because the code and class for the rule was already there before I started this PR, so I didn't think about removing it and instead tried to fix it. But if it can help remove complexity and our dance to have to fix the NPE for the `AppInitializer.kt` singleton being initialized too late etc… then maybe that's simpler indeed.

And once we'll migrate to Android 13 and the new way of doing per-app locale switching and get rid of all our logic for our custom in-app locale switching code that is around all that… we probably won't need any of this `WPLocaleTestRule` (since `WordPress.updateContextLocale(…)` and `AppInitializer.updateContextLocale()` will both have disappeared and rendered obsolete by us migrating to Android 13 and letting the OS handle that logic for us, right?)

Reply:
> So basically, remove the WPLocaleTestRule, and add a comment in the code of the test classes to tell developers to ensure they uninstall any previous version of the app before testing to ensure that the in-app locale switched and user preference for overridden locale does not accidentally take over the device locale set by the LocaleTestRule from fastlane?

Exactly! 💯 

> I'm ok with that tbh. Basically that fact of keeping WPLocaleTestRule came to be because the code and class for the rule was already there before I started this PR, so I didn't think about removing it and instead tried to fix it.

Yeap, I figured as much, I think it might be wise to take this chance and remove it now that we can (avoid having it there just for this case). Also, with this, we would potentially avoid the risk of adding to it later, for some unknown reason, thus risking making it indispensable and even harder to remove afterwards.

> But if it can help remove complexity and our dance to have to fix the NPE for the AppInitializer.kt singleton being initialized too late etc… then maybe that's simpler indeed.

Exactly! 💯 

> And once we'll migrate to Android 13 and the new way of doing per-app locale switching and get rid of all our logic for our custom in-app locale switching code that is around all that… we probably won't need any of this WPLocaleTestRule (since WordPress.updateContextLocale(…) and AppInitializer.updateContextLocale() will both have disappeared and rendered obsolete by us migrating to Android 13 and letting the OS handle that logic for us, right?)

True, but again, I think this might be a slight overstatement as us migrating to Android 13 doesn't mean that all of the devices are going to be utilizing Android 13. We would still need to support Android 12 and below for years to come. 🤔

However, if you mean that we can then update our emulators from our current usage of `API 28 (Android 9)` to using the newest `API 33 (Android 13)` for the screenshot generation, then yes, this is doable. 👍

Reply:
> However, if you mean that we can then update our emulators from our current usage of API 28 (Android 9) to using the newest API 33 (Android 13) for the screenshot generation, then yes, this is doable. 👍

Yep, that's what I had in mind.

For the record, we will very likely update the API level of the emulators used by the screenshot automation in a separate PR, from API 28 to at least API 31… if not 33, as part of fixing the "Gray overlay sometimes appears on views" bug mentioned by @zwarm in https://github.com/wordpress-mobile/WordPress-Android/issues/16900 that seems to only happen on API<30.

I just haven't updated those as part of this PR because that PR focused on making the automation work again in its current state, so preferred to keep our exploration of that gray overlay and API level bump in separate future PR; but if we update the API level of those emulators to API 33 as part of this future PR, that would be one more reason for us to get rid of `WPLocaleTestRule` in this current PR, considering that we will soon remove our in-app locale switcher on devices that are using Android 13 anyway.

Reply:
Got it, thanks so much for the clarification on that Olivier! 👍 

> ...from API 28 to at least API 31… if not 33

I would recommend going with `API 31` for now as we are not ready for `API 33` just yet and I believe, we will not be there for some time...